assets/data_table/main.js

```diff
@@ -106,14 +109,157 @@ function App({ ctx, data }) {
     content.page === content.max_page && !infiniteScroll
       ? totalRows % content.limit
       : content.limit;
 
+  const drawHeader = useCallback((args) => {
+    const {
+      ctx,
+      theme,
+      rect,
+      column,
+      menuBounds,
+      isHovered,
+      isSelected,
+      spriteManager,
+    } = args;
+
+    if (column.sourceIndex === 0) {
+      return true;
+    }
+
+    ctx.rect(rect.x, rect.y, rect.width, rect.height);
+
+    const basePadding = 10;
+    const overlayIconSize = 19;
+
+    const fillStyle = isSelected ? theme.textHeaderSelected : theme.textHeader;
+    const fillInfoStyle = isSelected ? theme.accentLight : theme.textDark;
+    const shouldDrawMenu = column.hasMenu === true && isHovered;
+    const hasSummary = column.summary ? true : false;
+
+    const fadeWidth = 35;
+    const fadeStart = rect.width - fadeWidth;
+    const fadeEnd = rect.width - fadeWidth * 0.7;
+
+    const fadeStartPercent = fadeStart / rect.width;
+    const fadeEndPercent = fadeEnd / rect.width;
+
+    const grad = ctx.createLinearGradient(rect.x, 0, rect.x + rect.width, 0);
+    const trans = withAlpha(fillStyle, 0);
+
+    const middleCenter = getMiddleCenterBias(
+      ctx,
+      `${theme.headerFontStyle} ${theme.fontFamily}`
+    );
+
+    grad.addColorStop(0, fillStyle);
+    grad.addColorStop(fadeStartPercent, fillStyle);
+    grad.addColorStop(fadeEndPercent, trans);
+    grad.addColorStop(1, trans);
+
+    ctx.fillStyle = shouldDrawMenu ? grad : fillStyle;
+
+    if (column.icon) {
+      const variant = isSelected
+        ? "selected"
+        : column.style === "highlight"
+        ? "special"
+        : "normal";
+
+      const headerSize = theme.headerIconSize;
+
+      spriteManager.drawSprite(
+        column.icon,
+        variant,
+        ctx,
+        rect.x + basePadding,
+        rect.y + basePadding,
+        headerSize,
+        theme
+      );
+
+      if (column.overlayIcon) {
+        spriteManager.drawSprite(
+          column.overlayIcon,
+          isSelected ? "selected" : "special",
+          ctx,
+          rect.x + basePadding + overlayIconSize / 2,
+          rect.y + basePadding + overlayIconSize / 2,
+          overlayIconSize,
+          theme
+        );
+      }
+    }
+
+    ctx.fillText(
+      column.title,
+      menuBounds.x - rect.width + theme.headerIconSize * 2.5 + 14,
+      hasSummary
+        ? rect.y + basePadding + theme.headerIconSize / 2 + middleCenter
+        : menuBounds.y + menuBounds.height / 2 + middleCenter
+    );
+
+    if (hasSummary) {
+      const summary = column.summary;
+
+      const numericInfo = {
+        Min: summary.min ?? "",
+        Max: summary.max ?? "",
+        Mean: summary.mean?.toFixed(2) ?? "",
+        Nulls: summary.nulls ?? "",
+      };
+
+      const categoricalInfo = {
+        Unique: summary.unique ?? "",
+        Top: summary.top ?? "",
+        "Top freq": summary.top_freq ?? "",
+        Nulls: summary.nulls ?? "",
+      };
```

Comment:
This looks great but I think we can move this to the server, what do you think?

This way, the server would send something like this:

```
[
  %{key: "Unique", value: "321"},
  ...
]
```

And all we do is to render the given keys and values? This means if we want to change how what is rendered, we just need to change the Elixir side and we don't have to write more JavaScript. :)